src/syntax/FStar.Syntax.Util.fs

```diff
@@ -250,6 +250,25 @@ let comp_to_comp_typ (c:comp) : comp_typ =
        flags=comp_flags c}
     | _ -> failwith "Assertion failed: Computation type without universe"
 
+
+(*
+ * For layered effects, given a (repr a is), return is
+ * For wp effects, given a (unit -> M a wp), return wp
+ *
+ * The pattern matching is very syntactic inside this function
+ *)
+let effect_indices_from_repr (repr:term) (is_layered:bool) (r:Range.range) (err:string)
+: list<term> =
+  let err () = Errors.raise_error (Errors.Fatal_UnexpectedEffect, err) r in
+  let repr = compress repr in
+  if is_layered
+  then match repr.n with
```

Comment:
Does the representation have to be an app node for a layered effect? I could use a arrow literal, right?
Also, maybe it is safer to use U.head_and_args in the this case, rather than directly matching on a Tm_app

Likewise, in the non-layered case, does the repr have to be an arrow? And to use arrow_formals_comp to destruct it, if it is an arrow.

Reply:
This function is expected to be called with the shape that it matches on (there is also a comment on the function). The idea is to be able to parse the effect indices given a `repr` term that is exactly `repr a is` for layered effects and `unit -> M a wp` for `wp` effects.

The way we store the layered effect combinator types in the env, including lifts, it is already the case that computation arguments (`f`, `g`, etc.) to those combinators have exactly these shapes.

Reply:
I added further comments to this function.